### PR TITLE
(EAI-233): Evaluation CLI docs expansion

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-233

## Changes

- Initial doc set for the evaluation CLI

## Notes

-

### CODE REVIEW

File: docs/docs/evaluation/index.md [hunk 1]
```diff
@@ -22,6 +13,24 @@ The CLI evaluates the chatbot's performance by comparing the chatbot's responses
 
 The Evaluation CLI works by running a series of commands that generate, evaluate, and report on the performance of your chatbot.
 
+Here's a high-level overview of the process:
+
+1. **Generate Data**: Generate data for evaluation. You can pull data from test cases files, databases, or APIs. Generated data outputs to a `GeneratedDataStore`.
+2. **Evaluate Data**: Evaluate the generated data. You can evaluate the quality, faithfulness, and other aspects of the data. Evaluation outputs to the `EvaluationStore`.
+3. **Report on Data**: Generate reports on an evaluation run. Reports output to the `ReportStore`.
+
+```mermaid
```
Comment (nlarew) [inline docs/docs/evaluation/index.md:22]: nit: consider moving this above the list. I think it's short enough that it doesn't push them below the fold + IMO it's more natural to have text below the diagram.

File: docs/docs/evaluation/command-reference.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# CLI Command Reference
+
+The MongoDB Chatbot Evaluation CLI has the following commands.
+
+You must first install the CLI before you can run it. See the [installation documentation](./index.md#install).
```
Comment (nlarew) [inline docs/docs/evaluation/command-reference.md:5]:
```suggestion
You must install the CLI before you can run it. See the [installation documentation](./index.md#install).
```

File: docs/docs/evaluation/configuration-reference.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
+# Configuration Reference
+
+This page contains reference documentation for the configuration options for the MongoDB Chatbot Evaluation CLI.
+
+A Evaluation CLI config files is a CommonJS file that exports a `ConfigConstructor` function as its default export.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:5]:
```suggestion
An Evaluation CLI config file is a CommonJS file that exports a `ConfigConstructor` function as its default export.
```

File: docs/docs/evaluation/configuration-reference.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+For an example of setting up a configuration file, refer to the [Configuration](./index.md) documentation.
+
+To set up a configuration file, you must first install the following packages:
+
+```bash
+npm install mongodb-chatbot-evaluation
+```
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:13]: Should we link out like on the other reference page? e.g. " You need to have the Evaluation CLI installed to configure it. See the installation docs."?

File: docs/docs/evaluation/configuration-reference.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+The [`ConfigConstructor`](../reference/evaluation/modules.md#configconstructor) function is the root configuration type for the Ingest CLI. This exports an [`EvalConfig`](../reference/evaluation/interfaces/EvalConfig.md) object.
+
+## Data Stores
+
+### `CommandMetadataStore`
+
+The [`CommandMetadataStore`](../reference/evaluation/interfaces/CommandMetadataStore.md) is an interface to interact with MongoDB collection that tracks metadata of each command run.
+
+To create an `CommandMetadataStore`, use the function [`makeMongoDbCommandMetadataStore()`](../reference/evaluation/modules.md#makemongodbcommandmetadatastore).
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:31]:
```suggestion
To create a `CommandMetadataStore`, use the function [`makeMongoDbCommandMetadataStore()`](../reference/evaluation/modules.md#makemongodbcommandmetadatastore).
```

File: docs/docs/evaluation/configuration-reference.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `GeneratedDataStore`
+
+The [`GeneratedDataStore`](../reference/evaluation/interfaces/GeneratedDataStore.md) is an interface to interact with MongoDB collection that stores generated data.
+
+To create an `GeneratedDataStore`, use the function [`makeMongoDbGeneratedDataStore()`](../reference/evaluation/modules.md#makemongodbgenerateddatastore).
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:46]:
```suggestion
To create a `GeneratedDataStore`, use the function [`makeMongoDbGeneratedDataStore()`](../reference/evaluation/modules.md#makemongodbgenerateddatastore).
```

File: docs/docs/evaluation/configuration-reference.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `ReportStore`
+
+The [`ReportStore`](../reference/evaluation/interfaces/ReportStore.md) is an interface to interact with MongoDB collection that stores reports.
+
+To create an `ReportStore`, use the function [`makeMongoDbReportStore()`](../reference/evaluation/modules.md#makemongodbreportstore).
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:76]:
```suggestion
To create a `ReportStore`, use the function [`makeMongoDbReportStore()`](../reference/evaluation/modules.md#makemongodbreportstore).
```

File: docs/docs/evaluation/configuration-reference.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+## `ConfigConstructor`
+
+The [`ConfigConstructor`](../reference/evaluation/modules.md#configconstructor) function is the root configuration type for the Ingest CLI. This exports an [`EvalConfig`](../reference/evaluation/interfaces/EvalConfig.md) object.
+
+## Data Stores
+
+### `CommandMetadataStore`
+
+The [`CommandMetadataStore`](../reference/evaluation/interfaces/CommandMetadataStore.md) is an interface to interact with MongoDB collection that tracks metadata of each command run.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:29]: Can we split this into two concepts?

- `CommandMetadataStore` is an interface for storing metadata every time a command is run
- We provide a constructor that implements `CommandMetadataStore` to store command metadata in MongoDB Atlas.

Same for the other stores

File: docs/docs/evaluation/configuration-reference.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+const reportStore = makeMongoDbReportStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+## Test Cases
+
+You must provide test cases to evaluate the chatbot. Pass the test cases to the `commands.generate` property in the `EvalConfig`.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:89]:
````suggestion
You must provide test cases to evaluate the chatbot. Pass the test cases to the `commands.generate` property in the `EvalConfig`.

```typescript
const testCases: ConversationTestCase[] = [
  {
    name: `It understands "why the chicken crossed the road" jokes`,
    expectation: `
      The ASSISTANT responds with a completion of the classic chicken crossing the road joke.
      The joke should be completed in a way that is both humorous and appropriate.
    `,
    tags: ["joke"],
    messages: [
      { role: "user", content: "Why did the chicken cross the road?" }
    ]
  },
];

const evalConfig: EvalConfig = {
  // ... other fields,
  commands: {
    generate: {
      myTest: {
        type: "conversation",
        testCases: testCases,
        generator: makeGenerateConversationData({ ... }),
      },
    },
    evaluate: { /* ... */ },
    report: { /* ... */ },
  },
};
```
````

I think this benefits from an up front example of what they look like / where they're defined.

File: docs/docs/evaluation/configuration-reference.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+
+The `mongodb-chatbot-evaluation` package includes built-in support for the [`ConversationTestCase`](../reference/evaluation/interfaces/ConversationTestCase.md) type.
+You can use this to evaluate the chatbot's performance on conversation data.
+
+You can load `ConversationTestCase` object from a YAML file using the [`getConversationsTestCasesFromYaml()`](../reference/evaluation/modules.md#getconversationstestcasesfromyaml) function.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:94]:
```suggestion
### Load test cases from a file
You can load `ConversationTestCase` object from a YAML file using the [`getConversationsTestCasesFromYaml()`](../reference/evaluation/modules.md#getconversationstestcasesfromyaml) function.
```

**Review by nlarew — COMMENTED** [review]
Nice work! A few small changes then will be good to go

File: docs/docs/evaluation/configuration-reference.md (hunk 8 above)
Comment (mongodben) [reply docs/docs/evaluation/configuration-reference.md:29]: updated

File: docs/docs/evaluation/configuration-reference.md [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+
+## `ConfigConstructor`
+
+The [`ConfigConstructor`](../reference/evaluation/modules.md#configconstructor) function is the root configuration type for the Ingest CLI. This exports an [`EvalConfig`](../reference/evaluation/interfaces/EvalConfig.md) object.
+
+## Data Stores
+
+### `CommandMetadataStore`
+
+The [`CommandMetadataStore`](../reference/evaluation/interfaces/CommandMetadataStore.md) is an interface to interact with MongoDB collection that tracks metadata of each command run.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:25]:
```suggestion
The [`CommandMetadataStore`](../reference/evaluation/interfaces/CommandMetadataStore.md) is an interface for storing metadata of each command run.
```

You could also do a flat file metadata store (or use another DB for w/e reason)

File: docs/docs/evaluation/configuration-reference.md [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+
+const commandMetadataStore = makeMongoDbCommandMetadataStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `GeneratedDataStore`
+
+The [`GeneratedDataStore`](../reference/evaluation/interfaces/GeneratedDataStore.md) is an interface to interact with MongoDB collection that stores generated data.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:43]:
```suggestion
The [`GeneratedDataStore`](../reference/evaluation/interfaces/GeneratedDataStore.md) is an interface for storing and working with generated evaluation data.
```

File: docs/docs/evaluation/configuration-reference.md [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+
+const evaluationStore = makeMongoDbEvaluationStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `ReportStore`
+
+The [`ReportStore`](../reference/evaluation/interfaces/ReportStore.md) is an interface to interact with MongoDB collection that stores reports.
```
Comment (nlarew) [inline docs/docs/evaluation/configuration-reference.md:79]:
```suggestion
The [`ReportStore`](../reference/evaluation/interfaces/ReportStore.md) is an interface for storing reports on the results of evaluation runs.
```

File: docs/docs/evaluation/pipeline.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+
+`runPipeline()` takes the following arguments:
+
+- `configConstructor`: A [`ConfigConstructor`](../reference/evaluation/modules.md#configconstructor) function that returns a configuration object.
+- `pipeline`: A [`Pipeline`](../reference/evaluation/modules.md#pipeline) function that defines the pipeline based on actions in the configuration.
+
+Example `runPipeline()` function call with its `ConfigConstructor` and `Pipeline`:
+
+```ts
+// somePipeLine.ts
```
Comment (nlarew) [inline docs/docs/evaluation/pipeline.md:21]:
```suggestion
```ts title="somePipeLine.ts"
```

Docusaurus code blocks have title support built in

**Review by nlarew — APPROVED** [review]
A few small tweaks, otherwise LGTM!

File: docs/docs/evaluation/pipeline.md (hunk 14 above)
Comment (mongodben) [reply docs/docs/evaluation/pipeline.md:21]: oh good to know. we use the same pattern employed here with the file name as a comment throughout a few places in docs so will keep same for consistency for now. but very open to change all going forward...good intern task. will make ticket

**Pull request merged**